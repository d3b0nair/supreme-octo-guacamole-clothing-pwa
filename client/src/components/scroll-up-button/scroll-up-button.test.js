import React from "react";
import {
  render,
  fireEvent,
  waitFor,
  screen,
  cleanup,
} from "@testing-library/react";
import ScrollUpButton from "./scroll-up-button.component";

describe("ScrollUpButton component", () => {
  const component = (mockProps) => <ScrollUpButton {...mockProps} />;
  window.scrollTo = jest.fn();
  afterEach(cleanup);

  it("should mount", () => {
    const { container } = render(component());
    expect(container).toMatchSnapshot();
  });
  it("should call window.scrollTo when container is clicked", async () => {
    render(component());
    await waitFor(() =>
      fireEvent.click(screen.getByTestId("ScrollUpButtonContainer"))
    );
    expect(window.scrollTo).toBeCalledWith(0, 0);
  });
});
