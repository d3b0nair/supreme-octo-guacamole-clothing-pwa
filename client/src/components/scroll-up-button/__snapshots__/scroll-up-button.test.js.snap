// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollUpButton component should mount 1`] = `
<div>
  <div
    class="sc-bdvvtL fpjVLv"
    data-testid="ScrollUpButtonContainer"
  >
    <svg>
      scrollUp.svg
    </svg>
  </div>
</div>
`;
